### PR TITLE
fixed some Optional String display & tap to dismiss the Keyboard

### PR DESCRIPTION
**Pull request by ChengLuffy**
35eb1e9 d075fcf
- tweet String will display with **Optional()** which we don't want.
- tap to dismiss the Keyboard will allow you click the buttons in view's bottom.
- fixed the only Warning in  Project_04.

3167ab7
- Fixed my mistake.

73a8ddd
- Fixed a crash when the image cannot be downloaded.

8e48fcb
- Add access request to photo alblum.

### CODE REVIEW

File: Project 02 - LoveTweet/LoveTweet/ViewController.swift [hunk 1]
```diff
@@ -77,5 +77,10 @@ class ViewController: UIViewController {
     alert.addAction(UIAlertAction(title: buttonTitle, style: UIAlertActionStyle.default, handler: nil))
     self.present(alert, animated: true, completion: nil)
   }
+    
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+        super.touchesEnded(touches, with: event)
```
Comment (soapyigu) [inline Project 02 - LoveTweet/LoveTweet/ViewController.swift]: why super.touchesEnded(:with:) here?

File: Project 04 - Todo/Todo/DetailViewController.swift [hunk 2]
```diff
@@ -99,6 +99,11 @@ class DetailViewController: UIViewController {
       todo?.date = todoDatePicker.date
     }
     
-    navigationController?.popToRootViewController(animated: true)
+    let _  = navigationController?.popToRootViewController(animated: true)
   }
+    
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+        super.touchesBegan(touches, with: event)
```
Comment (soapyigu) [inline Project 04 - Todo/Todo/DetailViewController.swift]: Actually we do not need this line right? :)

**Comment by ChengLuffy** [comment]
A habit of preventing problems from happening.

在 2017年2月8日，02:21，Yi <notifications@github.com<mailto:notifications@github.com>> 写道：


@soapyigu requested changes on this pull request.

________________________________

In Project 02 - LoveTweet/LoveTweet/ViewController.swift<https://github.com/soapyigu/Swift30Projects/pull/7#pullrequestreview-20579290>:

> @@ -77,5 +77,10 @@ class ViewController: UIViewController {
     alert.addAction(UIAlertAction(title: buttonTitle, style: UIAlertActionStyle.default, handler: nil))
     self.present(alert, animated: true, completion: nil)
   }
+
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+        super.touchesEnded(touches, with: event)


why super.touchesEnded(:with:) here?

________________________________

In Project 04 - Todo/Todo/DetailViewController.swift<https://github.com/soapyigu/Swift30Projects/pull/7#pullrequestreview-20579290>:

>    }
+
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+        super.touchesBegan(touches, with: event)


Actually we do not need this line right? :)

―

**Comment by soapyigu** [comment]
I see, then why call super.touchesEnded() in touchesBegan()？

**Comment by ChengLuffy** [comment]
mistake fixed.

**Comment by soapyigu** [comment]
Thank you!

**Pull request merged**